### PR TITLE
Gizmo improvements

### PR DESCRIPTION
**Pull request by bkeys**
I made some improvements to the gizmo in the editor. But I wanted to discuss using the built-in gizmo that is available in Godot 4 (if the plan is to move over there once a stable release has been made).
Either way I am excited for you to take a look!

### CODE REVIEW

File: src/Data/Scripts/Rail.gd [hunk 1]
```diff
@@ -53,7 +53,7 @@ var attached_signals: Array = []
 
 func _ready() -> void:
 	update_parallel_rail_settings()
-	manual_moving = false
+	manual_moving = true
```
Comment (HaSa1002) [inline src/Data/Scripts/Rail.gd]: Please set this value as default above

File: src/Editor/Scripts/Gizmo.gd [hunk 2]
```diff
@@ -11,7 +11,11 @@ var x_rot_active = false
 var mouseMotion = Vector2(0,0)
 func _input(event):
 	if event is InputEventMouseMotion:
-		mouseMotion = mouseMotion + event.relative
+		# Z will be reversed if we don't do this
+		if z_active:
+			mouseMotion = mouseMotion - event.relative
+		else:
+		  mouseMotion = mouseMotion + event.relative
```
Comment (HaSa1002) [inline src/Editor/Scripts/Gizmo.gd]:
```suggestion
		 	mouseMotion = mouseMotion + event.relative * (-1 if z_active else 1)
```

I don't think comments justifying changes is a good thing. Perhaps that is just me. What's definitely me is using a factor and a ternary if to reduce repition and line usage. If you don't follow that, please fix your indention issue on line 18.

File: src/authors.tres [hunk 3]
```diff
@@ -5,5 +5,5 @@
 [resource]
 script = ExtResource( 1 )
 developers = [ "Jean3219 - Jean28518", "Johannes Witt - HaSa1002", "Dominik Waurenschk - DasCapschen" ]
-contributors = [ "librebob", "Bohdan Matviiv - bodamat", "Michele Comignano - comick", "Matthew - skyace65", "Yann Dirson - ydirson", "IntinteDAO", "Uli Horn - ulihorn", "simfann", "SeaMac", "JupiterRider", "rogerm4242", "nalquas" ]
+contributors = [ "librebob", "Bohdan Matviiv - bodamat", "Michele Comignano - comick", "Matthew - skyace65", "Yann Dirson - ydirson", "IntinteDAO", "Uli Horn - ulihorn", "simfann", "SeaMac", "JupiterRider", "rogerm4242", "nalquas", "Brigham Keys, Esq. (bkeys)" ]
```
Comment (HaSa1002) [inline src/authors.tres]:
```suggestion
contributors = [ "librebob", "Bohdan Matviiv - bodamat", "Michele Comignano - comick", "Matthew - skyace65", "Yann Dirson - ydirson", "IntinteDAO", "Uli Horn - ulihorn", "simfann", "SeaMac", "JupiterRider", "rogerm4242", "nalquas", "Brigham Keys, Esq. - bkeys" ]
```
Please use a hyphen to split your real and nick name.

**Review by HaSa1002 — CHANGES_REQUESTED** [review]
I can't review the scene changes currently so here is just the code.

I forgot to mention the commit message style earlier. These are good, but please use imperative language. So "Add" instead of "Added" and "Pretty" instead of "Prettied". The reason being the commit message is releveant *before* the commit is applied. Meaning, you read the message as "If I apply this commit, I <commit message>". We do wanna have a capitalised first word for style, though :)

File: src/Data/Scripts/Rail.gd (hunk 1 above)
Comment (bkeys) [reply src/Data/Scripts/Rail.gd]: If we set `manual_moving` to false in the `_ready()` function of the rail then the will not be visible by default. How do you recommend achieving this?

Comment (HaSa1002) [reply src/Data/Scripts/Rail.gd]: remove the line there and set `manual_moving` to true in line 13

File: src/Data/UI/styles/complete_menu.tres [hunk 4]
```diff
@@ -65,8 +63,8 @@ Button/styles/focus = ExtResource( 18 )
 Button/styles/hover = ExtResource( 14 )
 Button/styles/normal = ExtResource( 16 )
 Button/styles/pressed = ExtResource( 13 )
-CheckButton/icons/off = ExtResource( 15 )
-CheckButton/icons/on = ExtResource( 4 )
+CheckButton/icons/off = null
+CheckButton/icons/on = null
```
Comment (nalquas) [inline src/Data/UI/styles/complete_menu.tres]:
```suggestion
CheckButton/icons/off = ExtResource( 15 )
CheckButton/icons/on = ExtResource( 4 )
```

File: src/Data/UI/styles/complete_menu.tres [hunk 5]
```diff
@@ -14,7 +13,6 @@
 [ext_resource path="res://Data/UI/styles/panel_section.tres" type="StyleBox" id=12]
 [ext_resource path="res://Data/UI/styles/button_right_pressed.tres" type="StyleBox" id=13]
 [ext_resource path="res://Data/UI/styles/button_right_hover.tres" type="StyleBox" id=14]
-[ext_resource path="res://Data/UI/check_button/off.png" type="Texture" id=15]
 [ext_resource path="res://Data/UI/styles/button_right_normal.tres" type="StyleBox" id=16]
```
Comment (nalquas) [inline src/Data/UI/styles/complete_menu.tres]:
```suggestion
[ext_resource path="res://Data/UI/check_button/off.png" type="Texture" id=15]
[ext_resource path="res://Data/UI/styles/button_right_normal.tres" type="StyleBox" id=16]
```

File: src/Data/UI/styles/complete_menu.tres [hunk 6]
```diff
@@ -1,9 +1,8 @@
-[gd_resource type="Theme" load_steps=29 format=2]
+[gd_resource type="Theme" load_steps=27 format=2]
 
 [ext_resource path="res://Data/UI/styles/line_edit_read_only.tres" type="StyleBox" id=1]
 [ext_resource path="res://Data/Fonts/FontMedium.tres" type="DynamicFont" id=2]
 [ext_resource path="res://Data/Fonts/FontIngame.tres" type="DynamicFont" id=3]
-[ext_resource path="res://Data/UI/check_button/on.png" type="Texture" id=4]
 [ext_resource path="res://Data/UI/styles/line_edit_normal.tres" type="StyleBox" id=5]
```
Comment (nalquas) [inline src/Data/UI/styles/complete_menu.tres]:
```suggestion
[ext_resource path="res://Data/UI/check_button/on.png" type="Texture" id=4]
[ext_resource path="res://Data/UI/styles/line_edit_normal.tres" type="StyleBox" id=5]
```

File: src/Data/UI/styles/complete_menu.tres [hunk 7]
```diff
@@ -1,9 +1,8 @@
-[gd_resource type="Theme" load_steps=29 format=2]
+[gd_resource type="Theme" load_steps=27 format=2]
```
Comment (nalquas) [inline src/Data/UI/styles/complete_menu.tres]:
```suggestion
[gd_resource type="Theme" load_steps=29 format=2]
```

**Review by nalquas — CHANGES_REQUESTED** [review]
Please revert your changes to `src/Data/UI/styles/complete_menu.tres`, they cause theming regressions in the settings menu.

File: src/Data/UI/styles/complete_menu.tres (hunk 4 above)
Comment (bkeys) [reply src/Data/UI/styles/complete_menu.tres]: I never even opened that file I don't know how it somehow got edited, reverting it now.

File: src/Data/UI/styles/complete_menu.tres [hunk 8]
```diff
@@ -170,4 +170,4 @@ WindowDialog/constants/title_height = 19
 WindowDialog/fonts/title_font = ExtResource( 2 )
 WindowDialog/icons/close = null
 WindowDialog/icons/close_highlight = null
-WindowDialog/styles/panel = ExtResource( 24 )
+WindowDialog/styles/panel = ExtResource( 24 )
```
Comment (HaSa1002) [inline src/Data/UI/styles/complete_menu.tres]: still not reverted

**Review by HaSa1002 — CHANGES_REQUESTED** [review]
Please clean the history into two commits. One for the gizmo changes and and one adding yourself as author. Additionally, please rephrase both. If you need help, feel free to ask here or on Discord.

**Comment by HaSa1002** [comment]
The code change is good for sure. The cylinders are a change. It definitely needs more radical changes but a proposal will follow in #166 I'll merge the changes as soon as the history is looking good :)

**Comment by bkeys** [comment]
Alright I cleaned up the git history, please review at your earliest convenience.

**Review by HaSa1002 — APPROVED** [review]
lgtm. I just did the edit myself and removed the one line ending change.

**Comment by HaSa1002** [comment]
CC @nalquas

**Review by nalquas — APPROVED** [review]
lgtm

**Comment by HaSa1002** [comment]
Thank you very much (and) for your patience and congrats to your first merged PR 🎉

**Pull request merged**